### ISSUE TITLE
[Feature Request] Sublists based on folder structure

### ISSUE THREAD

**Issue by chinagreenelvis** [libretro/RetroArch#13322]
I keep my ROMs sorted into subdirectories based on region, license, genre, etc. Rather than manually populating RetroArch with tons of custom playlists based on my categorizations, it would be much more convenient (and easier to browse) if the files added to any given playlist from a directory had the option of being sub-listed within that playlist, reflecting the folder structure.

**Comment by i30817** [comment]
What would be more convenient yet (if i wasn't convinced it was impossible because of lacking metadata for hacks) would be if you could save metadata searches (from 'explore') as playlists, with some kind of more robust search and more robust metadata (not just 'plain and uninformative genre, company and language')

**Comment by i30817** [comment]
That said your feature is a nice idea but you can kind of do it today with the manual scanner - it can autoupdate as you move games around because it saves the scanned directory.

**Comment by chinagreenelvis** [comment]
A feature for saving explore filters as custom playlists would be nice. In the meantime, allowing playlists to have the option of sublist structures based on the organization of the files themselves would solve my problems entirely since I don't have to wait on a community-driven metadata solution to organize my own ROMs.

**Comment by i30817** [comment]
The problem here is that this solution was already rejected for the manual scanner, a solution that is frankly superior. I even think, can just create a empty playlist with a prototype of what the manual scanner wants and 'refresh' the playlist to get get what you want, with the gotchas i mentioned.

**Comment by chinagreenelvis** [comment]
Why would the option for subfolder respect on a manual scan - and the ability to navigate playlists according to that folder structure - be rejected? I would think it would be a desirable _option_ for many users.

The most important thing for me, besides organization, is the navigation. I'm sure I could automate a script to write some manual playlists according to my categories and sub-categories, but the result would be an ugly mess since each playlist would wind up taking up a primary spot in the navigational menu - I would have to scroll through each one for every system, rather than just changing systems and then navigating by region/genre/playercount, etc.

Allowing for subfolder recognition on a scan, again - not by default but as an _option_, seems like a much more elegant way of addressing the problem, especially since metadata for games is functionally non-existent and will probably continue to be for many years to come.

**Comment by i30817** [comment]
Ah you want subfolders in the actual playlist. I was misunderstanding almost the whole thing myself.

Good luck with your FR then.

**Comment by chinagreenelvis** [comment]
Right; an optional subfolder structure within a primary playlist.

**Comment by agluck91** [comment]
Just want to plus one this one

**Comment by GreyFoxGer** [comment]
Still no subfolder possible?

**Comment by chinagreenelvis** [comment]
> Still no subfolder possible?

Technically, this is possible if you set your base content folder to a general ROMs directory and then use that for browsing the games you want to play. Personally, since starting this request I've moved onto using LaunchBox as a frontend... which unfortunately also does not have a straightforward way to create playlists based on folder structures LOL (it is however possible, but tedious and requires using manual playlist filters).

Either way, it's overall a better library management system, whereas RetroArch is the best one-stop-shop just for the emulation.

**Comment by i30817** [comment]
The manual scanner and the custom system name option can create seperate playlists for the same core with different subdirs by using different base scan dirs and naming them different (which names the playlist different). Indeed, I usually keep adventure game studio, interactive fiction and the rest of scummvm games in different base directories and playlists even if they all use the scummvm core (because the AGS games list is several hundreds and interactive fiction games are not graphical games, but text games and so completely different than graphics adventures in feel).

Unfortunately due to a misdesign during the creation of the original playlists\the playlist widget\the new manual scanner, using that (new to RetroArch at the time) option sabotages the normal RetroArch process to get the playlist system icon and thumbnails, because RetroArch wasn't originally designed to allow users to rename playlists and keep thumbnails - even if they could before by hand (I do have a alternative in my custom thumbnail downloader project https://github.com/i30817/libretrofuzz but it's a program outside RetroArch).

The problem is that "System Name", is NOT ONLY the playlist name. It's the database¹ filename, and thumbnail dir in the server and the thumbnail download dir too, and worse, it's not stored inside the playlist² but in the filename, so renaming a playlist after scanning it with one of the "canonical" system names won't work to preserve\further download thumbnails (except a tool that doesn't use RetroArch to create the dirs, like my own I linked above) and nothing will fix the system icons (the console icons you see on a playlist tab), because that is hard coded to the names by retroarch as a asset file, so you can't just dump a file in a "unexpected (to RetroArch)"  user directory to fix it like thumbnails. 

¹ (so metadata won't work, not that it does for manual playlists anyway - they don't have serials or checksums and the options that use the database don't have fallbacks to use name+system)
² (kinda, it's stored per game but not used for thumbnails outside of the history playlist for a incomprehensible reason, probably just something no one bothers to fix, because they're afraid to touch the c code they don't understand\fear will cause regressions like so many other things in RetroArch - I bet the reason it's stored in game entries is because the history playlist has games from multiple systems so they wanted a way to just copy the system\database string and core path at once into it)

tldr: RetroArch uses the playlist filename minus extension as a path component to find the database file, local downloaded thumbnail dir, and server remote thumbnail dir, in spite of not actually needing to do that because that information is inside each playlist game entry. This would be better if that info was a single "system name" property of the playlist and when constructing the history playlist entries they added that to the game entries, instead of writing them out with that extra (useless, unlike the core, which the user can change per game) entry (this is not without consequences, the C64 tosec playlist for example, is 20 mb or so, it could be shorter without the system name in each game entry). If that happened and the parts of RetroArch that still use the filename would be updated to use that property instead, the filename could be changed without affecting thumbnails or the database. Instead, this never happened. The problem became more notable when the manual scanner was created because it allows users to change the playlist database\thumbnail dirs (and thus its filename) from the scan options, worse the default of the option is "directory name" so if you don't name your directory scanned exactly the name of the database RA expects for the games in there, you get no thumbnails with the default option. Changing the playlist filename is actually useful, the database\thumbnail dirs to a custom one not so much. But both options could have been decoupled at the time by adding a new optional option "playlist name", change up "system name" to make it obligatory - but crucially without a default when opening the scan so the user is _forced_ to choose the correct one instead of the first _unless_ the user wrote a custom playlist name, then add a "None" option to "System name" (this is to accommodate RetroArch platforms\cores without databases\thumbnails) - and remove the custom and directory name options. That new property should be also in the non manual normal scanner playlists so those can also be renamed without affecting anything after the refactor. 

As a aside, part of the way my libretrofuzz project works is to download thumbnails from any server thumbnail dir the user chooses to any playlist filename thumbnail dir, which is how it bypasses RetroArch problems downloading. RetroArch then uses that playlist filename to find the thumbnail dir and the thumbs there and everything is fine (except downloading more thumbnails from inside RetroArch). I would have to change it if this was fixed to use that new property in playlists when saving thumbnails (or the first entry version because there is no difference), but frankly that would be a improvement because there would only be one canonical download even if game was repeated across different playlists.
